JustSaying/Messaging/IMessagePublisher.cs

```diff
@@ -1,13 +1,27 @@
-﻿using System.Threading;
+using System.Threading;
 using System.Threading.Tasks;
 using JustSaying.Models;
 
 namespace JustSaying.Messaging
 {
     public interface IMessagePublisher
     {
-        Task PublishAsync(Message message);
+        Task PublishAsync(PublishEnvelope message, CancellationToken cancellationToken);
+    }
+
+    public static class MessagePublisherHelpers
```

Comment:
If these are all extensions, the type name should be `(I)MessagePublisherExtensions`.